### DISCUSSION TITLE
Photo while rotating

### DISCUSSION THREAD

**Discussion by JoeKae** [YarostheLaunchpadder/MissionPlanner-to-Litchi#22] [General]
# Situation:

When the next waypoint is not in a straight line the aircraft will rotate while the photo is captured.
This happens in both trigger methods, distance interval and photo waypoints.
See example:
![example_rotation](https://user-images.githubusercontent.com/52297861/203363673-d1869765-769e-4d17-80e2-e7cb7b88f0b0.JPG)

# How can we compensate for this?

I did not test anything of this yet.

## Distance interval

1. Detect a waypoint that has a rotation component to it.
2. Add a stay for action that is long enough to prevent a rotation while capturing the image. (1 second maybe?)
3. Profit

## Photo waypoints

Either do the same as in Distance interval or just add a delay after every Take Photo action. The latter would prevent a possible blur if the aircraft starts moving before the capture is complete.

**Comment by YarosMallorca** [comment]
How does this happen? Why in your waypoint there is a TakePhoto command?
Isn't survey in mission planner based on distance in the first place?
I never had this happen to me.

**Reply by JoeKae** [reply]
This happened to me in distance-based mode as well as in camera control mode.
This example is using the camera control mode, hence the take photo command.
Since the first time i did mapping with litchi, i deleted the "spinning" images by hand. But it would be nice if they did not happen in the first place.

In the distance-based mode, the "rotation waypoint" has to match with an interval trigger for a "spinning" image to happen.
In the camera control mode, the images are taken at waypoints, so the "spinning" image is much more likely to happen.

I will test the proposed changes later today.

**Reply by JoeKae** [reply]
I could not reproduce it today.
The light conditions were perfect so the shutter speed was quite fast.
I'll try again when the light conditions are significantly worse.

**Reply by JoeKae** [reply]
Today i had mediocre light conditions so i did a test and i was able to reproduce it.

# Testflight

## Reference

![example_rotation_blur](https://user-images.githubusercontent.com/52297861/203820125-de4f6d52-712e-4efb-893d-9b81c493c4c6.JPG)

## Additional delay

![example_rotation_fixed](https://user-images.githubusercontent.com/52297861/203820260-c2095730-4f85-47ca-8de4-63488f78041f.JPG)


## Conclusion

The problem seems to be due to the fact that in low light conditions, the exposure of the image is not yet complete when the aircraft starts rotating. Therefore, the additional delay after the exposure fixes this problem. Since the duration of the delay depends on the lighting conditions, no exact recommendation can be made here. However, 1 second should be sufficient in all cases.

I also noticed a little less blur in the direction of movement on straight sections when a delay was added after the shot. Though, in this case, the blur was minimal.

### Example 3x digital zoom
![example_movement_blur](https://user-images.githubusercontent.com/52297861/203824212-71d8da88-0032-4572-b8ca-5a028adf1ab5.JPG)

While it is easy to reproduce this behaviour in the camera control mode (waypoint actions used for photo capture) it is most easy to fix in this mode as well.

When using the distance trigger mode (photo capture intervals between waypoints) it is a bit more complicated. Adding a  delay on waypoints that have a rotation component should fix the "spinning image". 
But fixing Blur in the movement direction is not possible with the delay action.

Here the only options are:
- faster shutter speed (better lighting conditions)
- slower flight speed
- higher altitude
